Prefabs/Locomotion/DestinationLocations/SharedResources/Scripts/DestinationLocation.cs

```diff
@@ -0,0 +1,238 @@
+        /// </summary>
+        protected RaycastHit hoverHit = new RaycastHit();
+
+        /// <summary>
+        /// Handles a new enter action.
+        /// </summary>
+        /// <param name="data">The data that is entering the location.</param>
+        [RequiresBehaviourState]
+        public virtual void Enter(SurfaceData data)
+        {
+            if (data.Transform == null || !SourceValidity.Accepts(data.Transform))
+            {
+                return;
+            }
+
+            IsHovered = true;
+            HoveringElements.Add(data);
+            if (HoveringElements.Count == 1)
+            {
+                HoverActivated?.Invoke();
+            }
+            Entered?.Invoke(GenerateHoverPayload(data));
+        }
+
+        /// <summary>
+        /// Handles a new exit action.
+        /// </summary>
+        /// <param name="data">The data that is exiting the location.</param>
+        [RequiresBehaviourState]
+        public virtual void Exit(SurfaceData data)
+        {
+            if (data.Transform == null || !SourceValidity.Accepts(data.Transform) || !HoveringElements.Contains(data))
+            {
+                return;
+            }
+
+            IsHovered = false;
+            HoveringElements.Remove(data);
+            Exited?.Invoke(GenerateHoverPayload(data));
+            if (HoveringElements.Count == 0)
+            {
+                HoverDeactivated?.Invoke();
+            }
+        }
+
+        /// <summary>
+        /// Handles a new select action.
+        /// </summary>
+        /// <param name="data">The data that is selecting the location.</param>
+        [RequiresBehaviourState]
+        public virtual void Select(SurfaceData data)
+        {
+            if (data.Transform == null || !SourceValidity.Accepts(data.Transform) || !HoveringElements.Contains(data))
+            {
+                return;
+            }
+
+            IsActivated = true;
+            Activated?.Invoke(GenerateSelectedPayload(data));
+
+            if (!EmitExitOnSelect)
+            {
+                return;
+            }
+
+            foreach (SurfaceData element in HoveringElements.ToArray())
+            {
+                Exit(element);
+            }
+            HoveringElements.Clear();
+        }
+
+        /// <summary>
+        /// Handles a new deselect action.
+        /// </summary>
+        [RequiresBehaviourState]
+        public virtual void Deselect()
+        {
+            IsActivated = false;
+            Deactivated?.Invoke();
+        }
+
+        /// <summary>
+        /// Generates the payload to emit on the hovering events of <see cref="Entered"/> and <see cref="Exited"/>.
+        /// </summary>
+        /// <param name="data">The data that is mutating the hover state.</param>
+        /// <returns>The data to emit.</returns>
+        protected virtual SurfaceData GenerateHoverPayload(SurfaceData data)
```

Comment:
Instead of `Generate` use `Create` as that seems more widely used in .NET API. (Also update the XML docs on this method to say `Creates` then.)